plugins/logitech-bulkcontroller/fu-logitech-bulkcontroller-device.c

```diff
@@ -423,6 +423,145 @@ fu_logitech_bulkcontroller_device_compute_hash(GBytes *data)
+	JsonArray *json_devices;
+	JsonNode *json_root;
+	JsonObject *json_device;
+	JsonObject *json_object;
+	JsonObject *json_payload;
+	g_autoptr(JsonParser) json_parser = json_parser_new();
+
+	/* parse JSON reply */
+	if (!json_parser_load_from_data(json_parser,
+					(const gchar *)decoded_pkt->data,
+					decoded_pkt->len,
+					error)) {
+		g_prefix_error(error, "error in parsing json data: ");
+		return FALSE;
+	}
+	json_root = json_parser_get_root(json_parser);
+	if (json_root == NULL) {
+		g_set_error_literal(error,
+				    G_IO_ERROR,
+				    G_IO_ERROR_INVALID_DATA,
+				    "did not get JSON root");
+		return FALSE;
+	}
+	json_object = json_node_get_object(json_root);
+	json_payload = json_object_get_object_member(json_object, "payload");
+	if (json_payload == NULL) {
+		g_set_error_literal(error,
+				    G_IO_ERROR,
+				    G_IO_ERROR_INVALID_DATA,
+				    "did not get JSON payload");
+		return FALSE;
+	}
+	json_devices = json_object_get_array_member(json_payload, "devices");
+	if (json_devices == NULL) {
+		g_set_error_literal(error,
+				    G_IO_ERROR,
+				    G_IO_ERROR_INVALID_DATA,
+				    "did not get JSON devices");
+		return FALSE;
+	}
+	json_device = json_array_get_object_element(json_devices, 0);
+	if (json_device == NULL) {
+		g_set_error_literal(error,
+				    G_IO_ERROR,
+				    G_IO_ERROR_INVALID_DATA,
+				    "did not get JSON device");
+		return FALSE;
+	}
+	if (json_object_has_member(json_device, "name"))
+		fu_device_set_name(device, json_object_get_string_member(json_device, "name"));
+	if (json_object_has_member(json_device, "sw"))
+		fu_device_set_version(device, json_object_get_string_member(json_device, "sw"));
+	if (json_object_has_member(json_device, "type"))
+		fu_device_add_instance_id(device,
+					  json_object_get_string_member(json_device, "type"));
+	if (json_object_has_member(json_device, "status"))
+		self->status = json_object_get_int_member(json_device, "status");
+	if (json_object_has_member(json_device, "updateStatus"))
+		self->update_status = json_object_get_int_member(json_device, "updateStatus");
+
+	return TRUE;
+}
+
+static gboolean
+fu_logitech_bulkcontroller_device_get_data(FuDevice *device, GError **error)
+{
+	FuLogitechBulkcontrollerDevice *self = FU_LOGITECH_BULKCONTROLLER_DEVICE(device);
+	g_autoptr(GByteArray) device_request = g_byte_array_new();
+	g_autoptr(GByteArray) decoded_pkt = g_byte_array_new();
+	g_autoptr(GByteArray) device_response = g_byte_array_new();
+	FuLogitechBulkcontrollerProtoId proto_id = kProtoId_UnknownId;
+
+	/* sending GetDeviceInfoRequest. Device reports quite a few matrix, including status,
+	 * progress etc */
+	device_request = proto_manager_generate_get_device_info_request();
+	if (!fu_logitech_bulkcontroller_device_send_sync_cmd(self,
+							     CMD_BUFFER_WRITE,
+							     device_request,
+							     error)) {
+		g_prefix_error(error,
+			       "error in sending buffer write packet for device info request: ");
```

Comment:
maybe change all the `error in`  prefixes to start with `failed to`  -- we know it's an error and we normally write these in past tense

Reply:
Fixed